docs/guides/modding-guide.rst

```diff
@@ -0,0 +1,248 @@
+
+...We then add a handle improvement to the listed product within our loop. ::
+
+    local new = df.itemimprovement_itemspecificst:new()
+    new.mat_type, new.mat_index = item.mat_type, item.mat_index
+    -- new.maker = outputItems[0].maker -- not a typical improvement
+    new.type = df.itemimprovement_specific_type.HANDLE
+    outputItems[productNumber - 1].improvements:insert("#", new)
+    -- break -- multiple handles, multiple "the handle is made from"s, so no break
+
+It's all a bit loose and hacky but it works, at least if you don't have multiple stacks filling up one reagent.
+
+Let's also make some code to modify the fire rate of the siege crossbow. ::
+
+    eventful.onProjItemCheckMovement[modId] = function(projectile)
+        if projectile.distance_flown > 0 then -- don't repeat this
+            return
+        end
+
+        local firer = projectile.firer
+        if not firer then
+            return
+        end
+
+        local weapon = df.item.find(projectile.bow_id)
+        if not weapon then
+            return
+        end
+
+        local multiplier = tonumber(customRawTokens.getToken(weapon.subtype, "TUTORIAL_MOD_FIRE_RATE_MULTIPLIER")) or 1
+        firer.counters.think_counter = math.floor(firer.counters.think_counter * multiplier)
+    end
+
+Now, let's see how we could make some "pegasus boots". First, let's define the item in the raws: ::
+
+    [ITEM_SHOES:ITEM_SHOES_BOOTS_PEGASUS]
+        [NAME:pegasus boot:pegasus boots]
+        [ARMORLEVEL:1]
+        [UPSTEP:1]
+        [METAL_ARMOR_LEVELS]
+        [LAYER:OVER]
+        [COVERAGE:100]
+        [LAYER_SIZE:25]
+        [LAYER_PERMIT:15]
+        [MATERIAL_SIZE:2]
+        [METAL]
+        [LEATHER]
+        [HARD]
+        [EXAMPLE_MOD_MOVEMENT_TIMER_REDUCTION_PER_TICK:5] custom raw token (you don't have to say this every time)
+
+Then, let's make a ``repeat-util`` callback for once a tick: ::
+
+    repeatUtil.scheduleEvery(modId, 1, "ticks", function()
+
+Let's iterate over every active unit, and for every unit, initialise a variable for how much we are going to take from their movement timer and iterate over all their worn items: ::
+
+    for _, unit in ipairs(df.global.world.units.active) do
+        local amount = 0
+        for _, entry in ipairs(unit.inventory) do
+
+Now, we will add up the effect of all speed-increasing gear and apply it: ::
+
+        if entry.mode == df.unit_inventory_item.T_mode.Worn then
+            amount = amount + tonumber((customRawTokens.getToken(entry.item, "EXAMPLE_MOD_MOVEMENT_TIMER_REDUCTION_PER_TICK")) or 0)
+        end
+    end
+    dfhack.units.addMoveTimer(-amount) -- Subtract amount from movement timer if currently moving
+
+Your first whole mod
+--------------------
+
+Now, you may have noticed that you won't be able to run multiple functions on tick/as event callbacks with that ``modId`` idea alone. To solve that we can just define all the functions we want and call them from a single function.
```

Comment:
This isn't exactly true. The mod author could attach as many handlers as they like to an event, each named differently, but with the mod name as a prefix. E.g. mymod-unitchecker, mymod-itemchecker, etc. I'm not sure if we need to recommend one approach over another.

Reply:
True true, I will reshape that.